test/core/networking/server_test.dart

```diff
@@ -0,0 +1,296 @@
+
+    // Wait for handler to be called and check
+    await received.awaitValue(Duration(seconds: 5), raiseOnTimeout: true);
+    received.expect(equals(true));
+
+    // Teardown
+    server.close();
+  });
+  test('Can respond to http request', () async {
+    final WebServer server = WebServer(
+      InternetAddress.loopbackIPv4,
+      0,
+      httpOnRequest: (HttpRequest req) {
+        req.response.statusCode = 200;
+        req.response.write('ANYTHING');
+        req.response.close();
+      },
+    );
+    await server.ready;
+
+    final http.Response response = await http.get(getUri(server, 'http'));
+
+    // Check response
+    expect(response.statusCode, equals(200));
+    expect(response.body.toString(), equals('ANYTHING'));
+
+    // Teardown
+    server.close();
+  });
+  test('Returns 404 if no handler set', () async {
+    final WebServer server = WebServer(
+      InternetAddress.loopbackIPv4,
+      0,
+    );
+    await server.ready;
+
+    final http.Response response = await http.get(getUri(server, 'http'));
+
+    // Check response
+    expect(response.statusCode, equals(404));
+    expect(response.body.toString(), equals(''));
+
+    // Teardown
+    server.close();
+  });
+  test('Can establish websocket connection', () async {
+    final VHook<bool?> received = VHook<bool?>(null);
+
+    final WebServer server = WebServer(
+      InternetAddress.loopbackIPv4,
+      0,
+      wsOnMessage: (ServerSession sess, dynamic msg) async {
+        received.set(true);
+      },
+    );
+    await server.ready;
+
+    final WebSocket client = await WebSocket.connect(getUri(server, 'ws').toString());
+    client.add('ANYTHING');
+
+    // Wait for handler to be called
+    await received.awaitValue(Duration(seconds: 5), raiseOnTimeout: true);
+    received.expect(equals(true));
+
+    // Teardown
+    client.close();
+    server.close();
+  });
+  test('Can close all WebSocket connections', () async {
+    final VHook<bool?> established = VHook<bool?>(null);
+    final VHook<bool?> closed = VHook<bool?>(null);
+
+    final WebServer server = WebServer(
+      InternetAddress.loopbackIPv4,
+      0,
+      wsOnMessage: (ServerSession sess, dynamic msg) async {
+        established.set(true);
+      },
+    );
+    await server.ready;
+
+    final ClientSession client = ClientSession(getUri(server, 'ws').toString());
+    client.onDone = (ClientSession _) async {
+      closed.set(true);
+    };
+    client.send('ANYTHING');
+
+    // Wait for connection to be established
+    await established.awaitValue(Duration(seconds: 5), raiseOnTimeout: true);
+    established.expect(equals(true));
+
+    // Close server
+    server.close();
+
+    // Wait for connection to be closed
+    await closed.awaitValue(Duration(seconds: 5), raiseOnTimeout: true);
+    closed.expect(equals(true));
+    waitFor(() => server.isEmpty, timeout: Duration(seconds: 5), raiseOnTimeout: true);
+  });
+  test('Can redirect websocket', () async {
```

Comment:
```suggestion
  });
  
  test('Can redirect websocket', () async {
```

An empty line between the tests would be neat

Reply:
I really dont need them, but if they please you ^^